apps/google-analytics-4/frontend/src/locations/Sidebar.tsx

```diff
@@ -1,17 +1,61 @@
-import React from 'react';
-import { Paragraph } from '@contentful/f36-components';
+import React, { useState, useEffect } from 'react';
+import { Paragraph, Spinner } from '@contentful/f36-components';
 import { SidebarExtensionSDK } from '@contentful/app-sdk';
-import { /* useCMA, */ useSDK } from '@contentful/react-apps-toolkit';
+import { useCMA, useSDK } from '@contentful/react-apps-toolkit';
 
 const Sidebar = () => {
   const sdk = useSDK<SidebarExtensionSDK>();
-  /*
-     To use the cma, inject it as follows.
-     If it is not needed, you can remove the next line.
-  */
-  // const cma = useCMA();
+  const cma = useCMA();
+  const [isLoading, setIsLoading] = useState(true);
+  const [credentials, setCredentials] = useState({});
 
-  return <Paragraph>Hello Sidebar Component (AppId: {sdk.ids.app})</Paragraph>;
+  // TODO: get field from app installation parameters
+  // sdk.parameters.installation.fieldId[sdk.contentType.sys.id]
+  const fieldId = 'slug';
+
+  useEffect(() => {
+    (async () => {
+      const req = {
+        method: 'POST' as const,
+        url: 'http://localhost:8080/dev/api/credentials', // TODO: configure lambda DNS and POST endpoint
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          key: fieldId,
+          value: sdk.entry.fields[fieldId].getValue(),
+        }),
+      };
+
+      // add request verification signing secret to request headers
+      const { additionalHeaders } = await cma.appSignedRequest.create(
+        {
+          appDefinitionId: sdk.ids.app!,
+        },
+        {
+          method: req.method,
+          headers: req.headers,
+          body: req.body,
+          path: new URL(req.url).pathname,
+        }
+      );
+      Object.assign(req.headers, additionalHeaders);
+
+      // fetch google analytics credentials for the current entry
+      fetch(req.url, req)
+        .then((res) => {
+          if (res.status === 403) throw new Error('Unauthorized'); // TODO: add error state rendering instead of throwing exception
+
+          return res.json();
+        })
+        .then((credentials) => {
+          setIsLoading(false);
```

Comment:
I guess we want to stop loading in the catch case as well, so I would put the `setIsLoading(false)` in the .finally part